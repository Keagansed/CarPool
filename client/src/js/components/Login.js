import React, { Component } from 'react';
import '../../css/App.css';
import 'bootstrap/dist/css/bootstrap.min.css'; 

import { getFromStorage } from '../utils/localStorage.js'
import "../utils/loginQuery.js";
import Profile from "./Profile.js"


class Login extends Component {
  constructor(props){
    super(props);

    this.state ={
      token:'',
    };
  }

  componentDidMount(){
    const obj = getFromStorage('sessionKey');
    if(obj && obj.token){
      //verify token
      const { token } = obj;
      fetch('/api/account/verify?token='+token)
       .then(res => res.json())
       .then(json => {
            if(json.success){
                this.setState({
                    token:token
                })
            }
       });
    }
  }

  render() {
    if(!this.state.token){
        return(
            <div className="container">
                <div className="row">
                    <div className="col-md-6 col-md-offset-3">
                        <h3>Sign Up{this.token}</h3> <hr/>
                        <form id="signUpSubmit">
	                        <div className="form-group">
	                            <label htmlFor ="signUpfname">First Name</label>
	                            <input type="text" className="form-control" id="signUpfname" placeholder="Enter first name"/>
	                        </div>
	                        <div className="form-group">
	                            <label htmlFor="signUplname">Last Name</label>
	                            <input type="text" className="form-control" id="signUplname" placeholder="Enter last name"/>
	                        </div>
	                        <div className="form-group">
	                            <label htmlFor="signUpemail">Email address</label>
	                            <input type="email" className="form-control" id="signUpemail" placeholder="Enter email"/>
	                        </div>
	                        <div className="form-group">
	                            <label htmlFor="signUpid">ID</label>
	                            <input type="text" className="form-control" id="signUpid" placeholder="Enter ID"/>
	                        </div>
	                        <div className="form-group">
	                            <label htmlFor="signUppass">Password</label>
	                            <input type="password" className="form-control" id="signUppass" placeholder="Enter password"/>
	                        </div>
	                        <button type="submit" className="btn btn-primary" >Submit</button>
                        </form>
                    </div>

                    <div className="col-md-6 col-md-offset-3">
                        <h3>Sign In</h3><hr/>
                        <form id="signInSubmit">
	                        <div className="form-group">
	                            <label htmlFor="signInemail">Email address</label>
	                            <input type="email" className="form-control" id="signInemail" placeholder="Enter email"/>
	                        </div>
	                        <div className="form-group">
	                            <label htmlFor="signInpass">Password</label>
	                            <input type="password" className="form-control" id="signInpass" placeholder="Enter password"/>
	                        </div>
	                        <button type="submit" className="btn btn-primary" >Submit</button>
                        </form>
                    </div>
                </div>
            </div>
        );
    }else{
       return(
            //~ <div className="container">
                //~ <h3>Account</h3>
                //~ <button type="submit" className="btn btn-primary" id="logOutSubmit">Log Out</button>
            //~ </div>
	    <Profile _id={this.state.token} />
        ); 
    }
  }
}

export default Login;
