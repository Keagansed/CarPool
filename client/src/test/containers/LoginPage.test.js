import React from 'react';
import Login from '../../js/containers/LoginPage';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

class MockStore {
    token = null; 
    loggedIn = false; 
    registered = false; 
    toggleError = false;
    lemail = ''; 
    lpassword = '';
    sFName = '';
    sLName = '';
    semail = '';
    sPass1 = '';
    sPass2 = '';
    sId = '';

    setToken = (token) => {
        this.token = token;
    };

    setLoggedIn = (success) => {
        this.loggedIn = success;
        this.lemail = '';
        this.lpassword = '';
    };

    setRegistered = (val) => {
        this.registered = val;
    };

    signup = () => {
        
        if(this.sPass1 !== this.sPass2) {
            console.log("Passwords do not match");
        }else{
            console.log("Successfully signed up");
            this.setRegistered(true);
        }
    };

    authenticate = () => {
        this.setRegistered(false);
        this.setToken("testtoken123");
        this.setLoggedIn(true);
    };

    logout = () => {
        this.setLoggedIn(false);
        this.setToken('');
    };

}

describe('Login Page Component', () => {
    let container, instance;
    const mockStore = new MockStore();

    beforeAll(() => {
        container = shallow(<Login store={mockStore} />);
        instance = container.dive().instance();
    });

    it('captures snapshot', () => {
        const renderedValue = renderer.create(<MemoryRouter><Login store={mockStore} /></MemoryRouter>).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });

    it('renders correctly', () => {
        expect(container.length).toEqual(1)
    });

    it('updates login email value correctly', () => {
        const event = {target: {value: "test@email.com"}};
        expect(instance.state.email).toEqual('');
        instance.updateLoginEmailValue(event);
        expect(instance.state.email).toEqual("test@email.com");
    });

    it('updates login password value correctly', () => {
        const event = {target: {value: "1234"}};
        expect(instance.state.password).toEqual('');
        instance.updateLoginPasswordValue(event);
        expect(instance.state.password).toEqual("1234");
    });

    it('checks correct state variables correctly', () => {
        expect(instance.canBeSubmitted()).toEqual(true);
    });

    it('check state variables correctly with incorrect email', () => {
        instance.state.email = "invalidemail";
        expect(instance.canBeSubmitted()).toEqual(false);
    });

    it('check state variables correctly with incorrect password', () => {
        instance.state.email = "test@email.com";
        instance.state.password = "";
        expect(instance.canBeSubmitted()).toEqual(false);
    });
});