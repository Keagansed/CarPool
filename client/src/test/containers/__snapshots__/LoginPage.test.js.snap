// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page Component captures snapshot 1`] = `
Array [
  <header
    className="MuiPaper-root-18 MuiPaper-elevation4-24 MuiAppBar-root-9 MuiAppBar-positionStatic-13 MuiAppBar-colorPrimary-16 Login-topNav-1"
  >
    <div
      className="MuiToolbar-root-45 MuiToolbar-dense-48 MuiToolbar-gutters-46 Login-toolbar-2"
    >
      <a
        href="/Landing"
        onClick={[Function]}
        style={
          Object {
            "color": "inherit",
            "textDecoration": "none",
          }
        }
      >
        <button
          aria-label="Back"
          className="MuiButtonBase-root-55 MuiIconButton-root-49 MuiIconButton-colorInherit-50"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-54"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-58"
              color={undefined}
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-67"
          />
        </button>
      </a>
      <h2
        className="MuiTypography-root-74 MuiTypography-title-80 MuiTypography-colorInherit-94"
      >
        The Iminsys Carpool Platform
      </h2>
    </div>
  </header>,
  <main
    className="Login-layout-3"
  >
    <div
      className="MuiPaper-root-18 MuiPaper-elevation2-22 MuiPaper-rounded-19 Login-paper-4"
    >
      <div
        align="center"
        className="MuiAvatar-root-100 Login-avatar-6"
      >
        <img
          alt={undefined}
          className="MuiAvatar-img-102"
          sizes={undefined}
          src="logo.png"
          srcSet={undefined}
        />
      </div>
      <h1
        className="MuiTypography-root-74 MuiTypography-headline-79"
      >
        Log In
      </h1>
      <form
        className="Login-form-7"
      >
        <div
          className="MuiFormControl-root-103 MuiFormControl-marginNormal-104 MuiFormControl-fullWidth-106"
        >
          <label
            className="MuiFormLabel-root-114 MuiFormLabel-required-119 MuiInputLabel-root-107 MuiInputLabel-formControl-108 MuiInputLabel-animated-111"
            data-shrink={false}
            htmlFor="email"
          >
            Email Address
            <span
              className="MuiFormLabel-asterisk-120"
            >
               *
            </span>
          </label>
          <div
            className="MuiInputBase-root-134 MuiInput-root-121 MuiInput-underline-125 MuiInputBase-formControl-135 MuiInput-formControl-122"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="on"
              autoFocus={undefined}
              className="MuiInputBase-input-144 MuiInput-input-129"
              defaultValue={undefined}
              disabled={false}
              id="email"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={true}
              rows={undefined}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="MuiFormControl-root-103 MuiFormControl-marginNormal-104 MuiFormControl-fullWidth-106"
        >
          <label
            className="MuiFormLabel-root-114 MuiFormLabel-required-119 MuiInputLabel-root-107 MuiInputLabel-formControl-108 MuiInputLabel-animated-111"
            data-shrink={false}
            htmlFor="password"
          >
            Password
            <span
              className="MuiFormLabel-asterisk-120"
            >
               *
            </span>
          </label>
          <div
            className="MuiInputBase-root-134 MuiInput-root-121 MuiInput-underline-125 MuiInputBase-formControl-135 MuiInput-formControl-122"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInputBase-input-144 MuiInput-input-129 MuiInputBase-inputType-147 MuiInput-inputType-132"
              defaultValue={undefined}
              disabled={false}
              id="password"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={true}
              rows={undefined}
              type="password"
              value=""
            />
          </div>
        </div>
        <button
          className="MuiButtonBase-root-55 MuiButtonBase-disabled-56 MuiButton-root-151 MuiButton-contained-162 MuiButton-containedPrimary-163 MuiButton-raised-165 MuiButton-raisedPrimary-166 MuiButton-disabled-171 MuiButton-fullWidth-176 Login-submit-8"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="-1"
          type="submit"
        >
          <span
            className="MuiButton-label-152"
          >
            Log In
          </span>
        </button>
        <div
          className="MuiFormControl-root-103 MuiFormControl-marginNormal-104 MuiFormControl-fullWidth-106"
        >
          <div>
            <p
              className="MuiTypography-root-74 MuiTypography-body1-83"
              onClick={[Function]}
            >
              Forgot Password?
            </p>
          </div>
        </div>
      </form>
    </div>
  </main>,
]
`;
